pkg/minikube/schedule/daemonize_unix.go

```diff
@@ -0,0 +1,80 @@
+// +build !windows
+
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedule
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/VividCortex/godaemon"
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+	"k8s.io/minikube/pkg/minikube/localpath"
+)
+
+// KillExisting kills existing scheduled stops
+func KillExisting(profiles []string) error {
+	for _, profile := range profiles {
+		file := localpath.PID(profile)
+		f, err := ioutil.ReadFile(file)
+		if os.IsNotExist(err) {
+			return nil
+		}
+		defer os.Remove(file)
+		if err != nil {
+			return errors.Wrapf(err, "reading %s", file)
+		}
+		pid, err := strconv.Atoi(string(f))
+		if err != nil {
+			return errors.Wrapf(err, "converting %v to int", string(f))
+		}
+		p, err := os.FindProcess(pid)
```

Comment:
Please check that the pid is minikube process before killing it. Otherwise this will kill random processes.

Reply:
When `minikube stop` daemonizes, it saves its own PID via `os.Getpid()` to a file, and that is the PID that is being killed here.

I'm not sure how a random process could be killed from this code, could you explain if you still think that's the case?